src/main/java/org/betonquest/betonquest/quest/event/spawn/Equipment.java

```diff
@@ -0,0 +1,80 @@
+package org.betonquest.betonquest.quest.event.spawn;
+
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.Instruction.Item;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.item.QuestItem;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Mob;
+import org.bukkit.inventory.EntityEquipment;
+import org.bukkit.persistence.PersistentDataType;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A wrapper for the equipment and drops of an entity.
+ *
+ * @param helmet     the helmet to equip the entity with
+ * @param chestplate the chest plate to equip the entity with
+ * @param leggings   the leggings to equip the entity with
+ * @param boots      the boots to equip the entity with
+ * @param mainHand   the main hand item to equip the entity with
+ * @param offHand    the off-hand item to equip the entity with
+ * @param drops      the items to drop when the entity dies
+ */
+public record Equipment(@Nullable QuestItem helmet, @Nullable QuestItem chestplate,
+                        @Nullable QuestItem leggings, @Nullable QuestItem boots, @Nullable QuestItem mainHand,
+                        @Nullable QuestItem offHand, Item[] drops) {
+
+    /**
+     * Adds the drops to the entity.
+     *
+     * @param entity  the entity to add the drops to
+     * @param profile the profile to get the drop amounts from
+     * @throws QuestRuntimeException if the variable could not be resolved
+     */
+    public void addDrops(final Entity entity, @Nullable final Profile profile) throws QuestRuntimeException {
```

Comment:
In general I would use the most minimal interface for the methods when entities are passed